datawig/utils.py

```diff
@@ -395,3 +397,58 @@ def sample_cartesian(sets: List,
     assert width == 1  # at the end of this procedure, the leaf index set should have width 1.
 
     return out
+
+
+### Functionality for label shift detection. Experimental. ###
+
+def synthetic_label_shift_simple(N, label_proportions, error_proba, covariates=None) -> pd.DataFrame:
+    """
+    Generate data with synthetic label shift and a single string feature column
+
+    :param N: Number of observations
+    :param label_proportions: Dirichlet vector with label proportions
+    :param error_proba: Probability of sampling from a different covariate distribution
+    :param covariates: list of covariate names, random strings by default
+    """
+
+    if covariates is None:
+        covariates = []
+        for i in range(len(label_proportions)):
+            covariates.append(''.join([random.choice(string.ascii_letters.lower()) for n in range(6)]))
```

Comment:
There's also a [`rand_str`](https://github.com/awslabs/datawig/blob/23deab456060bc393224002211207d0c9b0b915e/datawig/utils.py#L135) function for that in utils.

Reply:
good call, thanks.